src/parser.ts

```diff
@@ -111,9 +110,12 @@ export class Parser extends DiagnosticEmitter {
   dependees: Map<string, Source> = new Map();
 
   /** Constructs a new parser. */
-  constructor(program: Program) {
-    super(program.diagnostics);
-    this.program = program;
+  constructor(
+    public diagnostics: DiagnosticMessage[] = [],
+    public sources: Source[] = []
```

Comment:
Perhaps make this one a simple parameter as well, assigning in the constructor, so all the fields can be documented together above?

Reply:
Yep. This is fine. All set!